services/wireguard/service/service_unix.go

```diff
@@ -39,9 +40,16 @@ func NewManager(
 	location location.ServiceLocationInfo,
 	natService nat.NATService,
 	portMap func(port int) (releasePortMapping func()),
-	options Options) *Manager {
+	options Options,
+	portPool resources.PortSupplier,
+) *Manager {
 
-	resourceAllocator := resources.NewAllocator(options.PortMin, options.PortMax, options.Subnet)
+	portSupplier := portPool
+	if options.PortMin != 0 {
```

Comment:
we should probably just accept a portSupplier interface as a param. We then don't have to perform this check here and can just consume the port supplier.